Lib/idlelib/textview.py

```diff
@@ -28,52 +29,104 @@ def place(self, **kwargs):
         raise TclError(f'{self.__class__.__name__} does not support "place"')
 
 
-class TextFrame(Frame):
-    "Display text with scrollbar."
+def add_config_callback(config_callback):
+    """Class decorator adding a configuration callback for Tk widgets"""
+    def decorator(cls):
+        class WidgetWithConfigHook(cls):
+            def __setitem__(self, key, value):
+                config_callback({key: value})
+                super().__setitem__(key, value)
 
-    def __init__(self, parent, rawtext, wrap='word'):
```

Comment:
I think I understand why you removed 'wrap' from the signature of this class, but I think it is a mistake.  First, the current code to add scrollbars worked fine and I am rather sure it would work, with whatever minor adjustment, for the purposes of this PR.  Second, removing the parameter requires the addition of a new feature comprising about 50 lines of non-trivial code that I not only believe is not needed for this issue, but will never by needed by IDLE.  If you have evidence otherwise, it should be presented in a separate issue on scrollable widgets (which I think is needed anyway).

To put it another way, your clever code might be useful if we were writing a library for 3rd party use, but we are not.  I can't think of any reason why IDLE would want to change the wrap setting after the initial setting.  We certainly do not need to do so for this issue.

Since Texts default to wrap='char', which we are unlikely to want, I think 'wrap' should be required and not have a default value.

Reply:
You're right, the added complexity is likely unwarranted here. I'll revert this change.